### PR TITLE
Introducing profile scala-2.13 for development use

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- introducing `scala-2.13` for development use only
- adapt the common modules compilable on Scala 2.13, including
  - kyuubi-common
  - kyuubi-ctl
  - kyuubi-ha
- use separate compile options for Scala 2.13 , skipping deprecated scala compiler option `-Yno-adapted-args`

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -2124,6 +2124,57 @@
             </properties>
         </profile>
 
+        <!-- For development only, not generally applicable for all modules -->
+        <profile>
+            <id>scala-2.13</id>
+            <properties>
+                <scala.binary.version>2.13</scala.binary.version>
+                <scala.version>2.13.11</scala.version>
```
Comment (pan3793) [inline pom.xml:2132]:
```suggestion
                <scala.version>2.13.8</scala.version>
```

Comment (pan3793) [reply pom.xml:2132]: let's align with Spark master

Comment (bowenliang123) [reply pom.xml:2132]: SGTM.

**Comment by bowenliang123** [comment]
Verified modules compilable on Scala 2.13:
- mvn clean install -DskipTests -Pflink-provided,hive-provided,spark-provided -Pscala-2.13 -pl :kyuubi-common_2.13 -am

**Comment by bowenliang123** [comment]
Thanks, merged to master.

**Comment by bowenliang123** [comment]
Thanks, merged to master.

**Pull request closed without merging**